lib/gitlab/ci/templates/pipeline/AndroidTemplate.yml

```diff
@@ -100,35 +109,30 @@ testRelease:
   variables:
     BUILD_TARGET: testRelease
 
-
 instrumentationTests:
   rules: *debug_rules
   extends: .testBase
   variables:
     ANDROID_SDK_ROOT: "/sdk"
     ADB_EXECUTABLE: $ANDROID_SDK_ROOT/platform-tools/adb
+    BUILD_TARGETS: connectedDebugAndroidTest
   script:
-    - sdkmanager --version
-
-    - sdkmanager "platform-tools"
-
-    # Not necessary, but just for surity
-    - chmod +x ./gradlew
-
-    - ls -l $ANDROID_HOME/platform-tools || true
-
-    - $ADB_EXECUTABLE start-server
-    - $ADB_EXECUTABLE devices
-    - $ADB_EXECUTABLE connect "$ANDROID_EMULATOR_IP:$ANDROID_EMULATOR_ADB_PORT"
-    - $ADB_EXECUTABLE devices
-
-    - |+
-      for app_package_name in $APP_PACKAGE_NAMES_TO_FORCE_UNINSTALL
-      do
-        adb uninstall "$app_package_name" || true; 
-      done
-
-    - ./gradlew -Pci --console=plain connectedDebugAndroidTest
+    - |
+      for BUILD_TARGET in $BUILD_TARGETS
+        do
+          sdkmanager --version
+          sdkmanager "platform-tools"
+          ls -l $ANDROID_HOME/platform-tools || true
+          $ADB_EXECUTABLE start-server
+          $ADB_EXECUTABLE devices
+          $ADB_EXECUTABLE connect "$ANDROID_EMULATOR_IP:$ANDROID_EMULATOR_ADB_PORT"
+          $ADB_EXECUTABLE devices
+          for app_package_name in $APP_PACKAGE_NAMES_TO_FORCE_UNINSTALL
+            do
+              adb uninstall "$app_package_name" || true; 
+            done
+          ./gradlew -Pci --console=plain $BUILD_TARGET
+        done
   artifacts:
     when: always
     expire_in: 1 week
```

Comment:
Do the path reports (lines 135 - 141) change with multiple build targets?

Reply:
Good question. Yes, but line 141 will gather all flavored report paths because of the `./**/build`.